pkg/config/config.go

```diff
@@ -1395,6 +1395,14 @@ func findUnknownEnvVars(config Config, environ []string) []string {
 	for _, key := range config.GetEnvVars() {
 		knownVars[key] = struct{}{}
 	}
+	// Force ignore the system probe env vars because
+	// we don't want to have warning about unused env vars when loading both core and
+	// sysprobe config
+	// On processes other than system-probe this is a no-op since `SystemProbe` is never
```

Comment:
`SystemProbe` is actually initialized for all processes, just the file is not necessarily loaded. This will change when  the component framework is fully adopted. Regardless, the agent does read the `system-probe.yaml` config file (if present), to be able to communicate with system-probe.

Reply:
I think this is a positive change regardless. There are some scenarios (single container) where system-probe env vars are going to be present and we do not want to warn about them.

Reply:
Oh you are right, got confused by something else, will update the comment